omnigibson/simulator.py

```diff
@@ -240,10 +253,24 @@ def __init__(
             viewer_height=gm.DEFAULT_VIEWER_HEIGHT,
             device=None,
         ):
+            assert (
+                lazy.omni.isaac.core.utils.stage.get_current_stage() is None
+            ), "Stage should not exist when creating a new Simulator instance"
+
+            # Here we assign self as the Simulator instance and as og.sim, because certain functions
+            # called downstream during the initialization of this object will try to access og.sim.
+            # This makes that possible (and also introduces possible issues around circular dependencies)
+            assert og.sim is None, "Only one Simulator instance can be created at a time!"
+            og.sim = self
+
             # Store vars needed for initialization
             self.gravity = gravity
             self._viewer_camera = None
             self._camera_mover = None
+            self._render_on_step = True
+
+            self._floor_plane = None
+            self._skybox = None
```

Comment:
hmm very interesting, that means each scene might see a slightly different "angle" of the skybox?

Reply:
Yes I think so. Do you think this slight difference in rgb observation could cause any problems?

Reply:
I think it's probably fine? And perhaps we could also swap out the current skybox for a more neutral one, e.g.: the off-white plain background that @wensi-ai has been using?

Reply:
@hang-yin if we are not doing this now, please add a ticket to create per-scene skyboxes.